### ISSUE TITLE
[LightGBM] [Fatal] Sum of query counts is not same with #data

### ISSUE THREAD

**Issue by reemakumble** [microsoft/LightGBM#4664]
## Description
Getting the following error '[LightGBM] [Fatal] Sum of query counts is not same with #data'

## Reproducible example
I'm trying to run a GridSearch with LGBRanker using the following code-
param_grid={
    'colsample_bytree':[0.3,0.9],
    'max_depth':[3,6]
            } 

estimator = lg.LGBMRanker()
cv = ShuffleSplit(n_splits=10, test_size=.2, random_state=0)

model = GridSearchCV(estimator=estimator, cv=cv, param_grid=param_grid, n_jobs=8,scoring='neg_mean_squared_error')
model.fit(X_train, y_train,group=groups)
print ("Best Estimator learned through GridSearch") 
print (model.best_estimator_)

## Environment info

Initially code was working for version - 3.2.1 now installed 3.3.0

Command(s) you used to install LightGBM

pip install lightgbm

## Additional Comments
Code was previously running fine, now is throwing a fatal error.
![lgbm](https://user-images.githubusercontent.com/92202348/136648119-600625b5-1f66-4e10-96fa-29acbbe7954a.JPG)

**Comment by StrikerRUS** [comment]
@reemakumble Thanks for using LightGBM!
Unfortunately, scikit-learn doesn't support ranking applications, therefore `LGBMRanker` cannot be successfully used with all tools from the sklearn ecosystem: #3018, #4243.
